src/l500/l500-color.cpp

```diff
@@ -485,6 +499,62 @@ namespace librealsense
     }
 
 
+    void l500_color_sensor::set_default_controls_for_calibration_if_needed()
+    {
+        for (auto && calib_control : _calib_controls)
+        {
+            auto && control = get_option(calib_control.option);
+
+            auto curr_val = control.query();
+            if (curr_val != calib_control.default_value)
+            {
+                LOG_INFO( "Calibration changed option: " << rs2_option_to_string( calib_control.option )
+                                              << " from: " << curr_val
+                                              << " to: " << calib_control.default_value );
+                calib_control.need_to_restore = true;
+                calib_control.previous_value = curr_val;
+                control.set(calib_control.default_value);
+            }
+        }
+    }
+
+    void l500_color_sensor::restore_pre_calibration_controls_if_needed()
+    {
+        for (auto && calib_control : _calib_controls)
+        {
+            auto && control = get_option(calib_control.option);
+
+            auto curr_val = control.query();
+            if( calib_control.need_to_restore &&
+                ( curr_val == calib_control.default_value ) )
+            {
+                LOG_INFO("Calibration restored option: " << rs2_option_to_string(calib_control.option)
+                    << " from: " << curr_val
+                    << " to: " << calib_control.previous_value);
+                control.set(calib_control.previous_value);
+            }
+            calib_control.need_to_restore = false;
+        }
+    }
+
+    void l500_color_sensor::register_calibration_controls()
+    {
+        _calib_controls.push_back({ RS2_OPTION_ENABLE_AUTO_EXPOSURE });
```

Comment:
This is using the aggregate initialization, then? But there are no default values for the rest of the struct members... what will be their values?

Even if this is correct, I think it's confusing enough to modify to make it clearer what the intent is.

I also do not understand why separate this from read_calibration_controls_defaults() -- shouldn't we initialize the values when we create them?

Reply:
Refactored as recommended